### ISSUE TITLE
There doesn't seem to be a way to set requires_grad=True for parameters you don't want to update

### ISSUE THREAD

**Issue by import-antigravity** [PyTorchLightning/pytorch-lightning#5368]
Environment:
macOS
Python 3.8
pytorch-lightning 1.0.4

I know the title seems kind of self-contradictory, but I'm trying to train a GAN where I'm regularizing the loss with the gradient norm like in [this paper](https://arxiv.org/pdf/1705.10461.pdf). I'm unable to calculate the gradient penalty correctly because, for example, `requires_grad` for the discriminator is set to false when I'm evaluating the generator, even though I need to calculate gradients through both.

I've already tried setting up a third optimizer to handle the regularization, but it may lead to inaccurate results because the regularizer is optimizing separately, as opposed to being added to the loss function.

If anyone has been able to implement something like this with lightning, I would really like to know. This issue has been giving me grief for weeks at this point and I haven't been able to figure out a solution.

**Comment by awaelchli** [comment]
perhaps it's easier to do this in the manual optimization mode?

**Comment by awaelchli** [comment]
https://pytorch-lightning.readthedocs.io/en/stable/optimizers.html#manual-optimization

**Comment by import-antigravity** [comment]
I believe it’s the same in manual optimization mode, requires_grad still gets set to False, you just ignore optimizer_idx. Unless I’m misunderstanding the documentation...

**Comment by awaelchli** [comment]
In manual mode, you need to handle your multiple optimizers yourself, calling step etc. You can do (almost) whatever you want, including setting `requires_grad` on the weights you want, but you can also make use of the `no_grad` context manager.

**Comment by import-antigravity** [comment]
Ah, I figured it out. I needed to set `manual_optimization=True` in the Trainer. Is there a way to specify Trainer arguments in the LightningModule class, for better encapsulation?

**Comment by awaelchli** [comment]
```python
@property
def automatic_optimization(self) -> bool:
    return False
```
in LightningModule.
Generally, Trainer is independent of model, so no.
